Bunch of extra draft ideas for the base template

camerons opened the pull request:
> Associated issue: https://github.com/thegooddocsproject/templates/issues/183

## Purpose / why

A bunch of suggested additions to the base template.

## What changes were made?

Refer to https://github.com/thegooddocsproject/templates/issues/183 for summary of changes.

## Verification

No verification required.
But I'm really hoping for extensive review from all our tech writers, as this base template will drive how all our other templates will be written.

---

## Checklist

> __Pull-request reviewer__ should ensure the following

* [ ] Are issues linked correctly?
* [ ] Is this PR labeled correctly?
* [ ] If template updates: do they align with [developers.google.com/style/](https://developers.google.com/style/)?
* [ ] Did the PR receive at least one :+1: and no :-1: from core-maintainers? *This is not required for our incubator repository.*
* [ ] On merging, did you complete the merge using [keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-to-an-issue)?
* [ ] On merging, did you add any applicable notes to a [draft release](https://github.com/thegooddocsproject/templates/releases) and link to this PR?

aidanatwork commented:
Sorry I haven’t replied yet— definitely plan to review the templates stuff this week or this weekend. Won’t have time today though.

Sent from my iPhone

> On Jan 6, 2021, at 2:56 PM, Cameron Shorter <notifications@github.com> wrote:
> 
> ﻿
> @camerons requested your review on: #18 Bunch of extra draft ideas for the base template.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub, or unsubscribe.

camerons commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+* [Currently implemented doctypes templates](https://github.com/thegooddocsproject/templates): To help identify if you need to create a new template or edit an existing one.
+* Technical writing: To produce a well-written document that includes the relevant references.
+
+## Target audience for templates
```
On Tue, Jan 5, 2021 at 8:46 AM S... wrote:
Hallo Cameron

Thanks - I had a good read. That's coming along very nicely! My one suggestion would be to reduce the number of audience types, if possible. Simpler is better, even if it means sacrificing some nuance. For example, maybe you don't need the two "reviewer" roles and the "doc reader" role. They're important roles, but it's not clear to me how a template would include tips for those roles.

Cheers

S

viraji replied:
I agree with S in that we can reduce the number of audience types and thereby simply the base template. 
Here’s what I suggest:

**Template author** -  merging the roles of template author, reviewer and customizer into one, because essentially, all these roles will access the Base Template with the requirement of editing/authoring a new template. 


**Document Reader** - from the point of view of anyone accessing the Base template, we can simplify the tips just for a reader and an author (above).

ryanmacklin replied:
+1 to that. If you're looking for another way to collapse roles, you could have "Template creator" to emcompass the three subroles listed here, "Docset owner" (if it needs to be separately called out), and "Document creator." The TA for the template isn't the reader, but what's produced is for the reader. Since you have a user-facing TA section later, having that called out here might conflate this for some readers?

camerons replied:
Thanks for the suggestions, I agree it improves things. I've trimmed the roles back.

camerons commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+And of course, we need to understand the needs of the final document reader. There are likely multiple reader personas to consider, which depends upon the reader’s role, their technical expertise, their specific goal, and so on.
+
+## Inheritance of template sets
```
On Thu, Jan 7, 2021 at 4:44 AM T... wrote:
Hi Cameron, I checked out the base template. From what I understand, this is a template to provide a starting point for other templates? That's very meta but I can see the point. Overall, the sections were a bit confusing for me to follow. It wasn't clear to me which sections were intended as part of the template inheritance or not. For example, the "Inheritance of Template Sets" section -- is this part of the foundation for a template or just high-level information about templates, etc? The sections need to be clearer, in my opinion.

camerons replied:
Simplified personas, and split the docs to help make things easier to understand.

camerons reviewed (commented):
Adding feedback I've received by email from S... and T...

viraji commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
 # About Base Template
```
+1 for this About Base Template description.

viraji commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
 # About Base Template
 
+{Template author tip:
```
+1 for Template author tip.

viraji commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
 
 ## Prerequisites
```
+1 for the revised Prerequisites.

viraji commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+_{TBD: Insert hierarchy diagram here.}_
+
+### Base (abstract) template set
```
+1 for Base template set description.

viraji commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+* base-template.md: Provides common structure, elements and writing tips for downstream xxx-template.md files.
+* about-base-template.md: Provides common structure, elements and writing tips for downstream about-xxx-template.md files.
+
+### Core (abstract) templates set
```
+ for Core templates set description.

camerons replied:
Hi @viraji , I'm unclear if you are suggesting a specific action here? If so, I'm unclear on what needs to change.
I'm assuming you are suggesting a +1, as per some of your other comments, and will close no action.
Please re-introduce your comment (with a specific action) if I've missed your point.
Thanks.

viraji commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+* If any of the microcontent on a site is inaccurate or dated, it will reduce the trustworthiness of all the microcontent on the site, as the user typically won’t be able to determine which of the content is trustworthy.
+* As such, a project should only adopt microcontent fields which are automatically maintained by your documentation publishing system, or where there is an established, low-effort process to ensure the microcontent is maintained.
+
+## Primary goal
```
+1 for stating down the primary goal. I tried doing this with the intro, but this is way more direct and concise! 👍🏼

viraji commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+{Template author tip: Concisely state the primary purpose of this doctype, ideally in one short sentence.}
+
+## Target audience and user stories
```
+1 for Target audience and user stories section. I believe descriptive examples of the use case also fits in here?

viraji commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+* As a {user persona}, I want to {achieve a specific goal}, so that I can {achieve a larger business objective}.
+    * {Docset owner tip: Include this user story when … .}
+
+## Nature of content
```
+for nature of content.

viraji commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+}
+
+## Priorities for each quality criteria
```
This section  (Priorities for each quality criteria) needs further refinement in my opinion. We can, for instance, better present the quality criteria and also guide the template author based on their priorities.

camerons replied:
Yes @viraji , I agree. I have a "TBD (To Be Decided)" comment in this section to remind us to come back to this.
I'll close this comment for the moment, and we can come back to it later.

viraji reviewed (approved):
I’ve gone through the changes in detail and I’m **+1** on all sections for a first review. 👍🏼 
Thank you @camerons   for the extensive changes done to the original template - I specially like the new headings that have been introduced. They convey the intended purpose in a much more direct way than just through the Intro descriptions we had.

flicstar commented on base-template/base-template.md:
```diff
@@ -1,77 +1,123 @@
 
-1. Technical documentation types - to be able to identify the type of documentation you need to create a template for, or modify the existing one into.</li>
-2. Writing technical documentation - to accurately and clearly express instructions to the reader.</li>
+* All templates must include snippets of [structured data](https://developers.google.com/search/docs/guides/sd-policies). to help search engines and facilitate content reuse. 
```
```suggestion
* All templates must include snippets of [structured data](https://developers.google.com/search/docs/guides/sd-policies) to help search engines and facilitate content reuse. 
```
And is the "must" really necessary? What if the author has no need for snippets?

flicstar commented on base-template/base-template.md:
```diff
@@ -1,77 +1,123 @@
-# Base template
+* This base template captures the structure and common elements that all doctype templates should follow. It provides guidance for template authors to follow when creating a new template.
+* “Template author and template reviewer tips” should be removed from the final {doctype}-template.md. Other tips should remain.
+* Tweak other tips to align with the doctype you are writing for.
```
```suggestion
* Adjust other tips to align with the doctype you are writing for.
```
or "modify".
"Tweak" has a drug connotation in the USA.

Loquacity replied:
I would ditch the 'for' at the end of the sentence, too.

camerons replied:
Thanks updated as suggested.

flicstar commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
+* This about-base-template captures the structure and common elements that all downstream about-xxx-templates should follow.
+* It provides guidance for both template authors as well as document authors.
+* “Template author and template reviewer tips” should be removed from the final about-{doctype}-template.md. Other tips should remain.
+* Tweak other tips to align with the doctype you are writing for.
```
```suggestion
* Adjust other tips to align with the doctype you are writing for.
```
or "modify"
Tweak has drug connotations in the USA

camerons replied:
Thanks, fixed.

flicstar commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
+
+}
+
+This document provides guidance related to the authoring of documents based on the {doctype}-template. It focuses on the “how”, “why” and where applicable, references background theory to justify the approaches recommended.
```
```suggestion
This document provides guidance related to the authoring of documents based on the {doctype}-template. It focuses on the “how” and “why”; and where applicable, references background theory to justify the approaches recommended.
```
How, why and where tend to go together - but that's not what we've got here, so we need to separate the "where" a bit more.

Loquacity replied:
smart quotes. They're all through, so I'm not going to correct them all.

camerons replied:
A bunch or rewording, and this section has been removed. I think I've fixed whatever quotes we had before.

flicstar commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
-This document is used to create a uniform, regulated template for producing multiple similarly structured documents, that is, as a base template to create any other template. The base template guides you through the steps to create a functional template that can be reused, adhered to, and followed consistently throughout.
+_{TBD: Introduction text below needs tweaking to align with revised base template structure.}_
+
+This document helps create a uniform, regulated template for producing multiple similarly structured documents, that is, as a base template to create any other template. The base template guides you through the steps to create a functional template that can be reused, adhered to, and followed consistently throughout.
```
Hmmm, uniform and regulated and similarly structured seem to be synonyms for template. I wonder if we can get away with less?
```suggestion
This document is a base template that you can use to create any other template. The base template guides you through the steps to create a functional template that can be reused, adhered to, and followed consistently throughout.
```

flicstar commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
+
+This document helps create a uniform, regulated template for producing multiple similarly structured documents, that is, as a base template to create any other template. The base template guides you through the steps to create a functional template that can be reused, adhered to, and followed consistently throughout.
 
 Using a well-defined, good template helps to maintain consistency and integrity across all documentation while simplifying the process of creating any required document. It can save you time and effort, while ensuring consistency, clarity, and uniformity across your documentation suite.
```
```suggestion
Using a well-defined template helps to maintain consistency and integrity across all documentation while simplifying the process of creating any required document. It can save you time and effort, while ensuring consistency, clarity, and uniformity across your documentation suite.
```

camerons replied:
Thanks, accepted.

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+* [Git](https://git-scm.com/): The version control system used.
+* [GitHub](https://github.com/thegooddocsproject): The project hosting platform used, along with associated project management tools.
+* [GitHub Flavored Markdown](https://github.github.com/gfm/): The wiki format used to store template documentation.
```
```suggestion
* [GitHub Flavored Markdown](https://github.github.com/gfm/): The markup language used to write template documentation.
```

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+* Select a set of doctype templates for the project.
+* Customize templates if needed.
+* Select a style guide for the project.
+* Establish a docset maintenance processes.
```
```suggestion
* Establish the docset maintenance process.
```

Loquacity replied:
If you're going to use the definite article here, you should use it in the previous points also.

camerons replied:
@Loquacity or @flicstar or someone else, I'm happy to take your advice here, and accept changes that you propose. As an impostor tech writer who's English was only just good enough to get into Engineering, my radar signal for picking up these grammatical details is pretty weak.

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+* {doctype}-template.md:
+    * Provides document headings, structure, common text, and specific writing tips for a specific &lt;doctype>.
+    * Writing tips focus on the “what” to include and not the “how”, “why”.
```
```suggestion
    * Writing tips focus on the “what” to include and not the “how” or the “why”.
```

camerons replied:
removed this text in later version.

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+    * Provides document headings, structure, common text, and specific writing tips for a specific &lt;doctype>.
+    * Writing tips focus on the “what” to include and not the “how”, “why”.
+* about-{doctype}-template.md:
+    * Provides further guidance related to the use of the template. It focuses on the “how”, “why” and where applicable, references background theory. It covers topics like:
```
```suggestion
    * Provides further guidance related to the use of the template. It focuses on the “how” and “why”; and where applicable, references background theory. It covers topics like:
```

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+## Microcontent
+
+_TBD: We need to decide on the correct term to use between: structured data, microcontent, micro-content, microdata, snippets, metadata._
```
Yeah right. This isn't what I thought it was at all. I would call this metadata.

ryanmacklin replied:
My thought as well. When I see "microcontent," my mind first goes to reusable snippets.

Loquacity replied:
:+1: to metadata

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+{Template author tip: 
+
+* Discuss the nature of the content for this doctype.
```
"nature of the content" isn't plain English so I didn't immediately understand what it meant. Do we mean "format" ?

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+{Template author tip: 
+
+* Discuss the nature of the content for this doctype.
+* Consider how it will impact maintenance.
+* Is this content expressed best with multi-media, such as diagrams, screenshots or video?
+* If so, reference your style guide which should set recommendations on image and screenshot size, video length, style etc.
```
```suggestion
* Decide what format will best express the nature of the content. For example, text only, or multi-media options such as diagrams, screenshots or video.
* Consider how the format you choose will impact maintenance.
* Reference your style guide which should have recommend guidelines for using multi-media such as image size and video length.
```

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+{Template author tip: 
+
+* Each doctype addresses different use cases. It target’s different user personas, with different needs.
```
```suggestion
* Each doctype addresses different use cases. It targets different user personas, with different needs.
```

flicstar commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+|Currency: Alignment with a specified version of software.||
+|Comprehensive: Covers every feature and use case.||
+|Well written: Unambiguous.||
+|Well written: Concise, high signal to noise ratio, high information density.||
```
Dunno what "high signal to noise ratio" means.

camerons replied:
Yeh, good catch. I'll delete reference to it.
It is a term used in relation to old school radio or TV signals. A geeky engineering concept.
Good signal, low static means you get a clear audio or picture.
Low signal means that you mostly get fuzzy static.

flicstar reviewed (approved):
I made some comments and suggestions. I got a bit bored toward the end there. I'm sure others will have more to say. I think this is a great start! Thanks so much.

ryanmacklin commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+## Target audience and user stories
+
+{Template author tip: You should document the key user stories that this template is likely to be used for.}
```
As I'm reading these tips, they look more like to-dos than suggestions or advisories. Why did you land on the word "tip" for the collection of these?

camerons replied:
@ryanmacklin I picked "tips" based on another template I'd seen (where the comments were actually tips rather than instructions), and also "tip" is a concise word. I agree that "tips" is probably not the write word here. I don't think "to-dos" is quite right either. Maybe:
"Template author instruction:" or
"Template author guide:" or
"Template author direction:"

ryanmacklin commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+This {doctype} template aims to address the following user stories:
+
+* As a {user persona}, I want to {achieve a specific goal}, so that I can {achieve a larger business objective}.
```
Cameron, I suspect this is particularly where you want my input. Let's have a conversation about where the emotional persona element could intersect here, and if there are other ways to construct user stories.

camerons replied:
As discussed, I'm looking for @ryanmacklin , Chris Ward, Aaron Peters, and others to take the lead on fleshing out this section in further iterations. I'll close this comment for the moment on the understanding that it will be addressed in a separate thread.

ryanmacklin commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+|--- |--- |
+|Currency: Alignment with the latest version of software.|{Select one of: Must, Should, Nice to have (May), Not Applicable (N/A), Should not}|
+|Currency: Alignment with a specified version of software.||
+|Comprehensive: Covers every feature and use case.||
```
Would "Goes beyond the 'happy path'" fit as "comprehensive"?

camerons replied:
@ryanmacklin , I think you are onto something here. I think "happy path" is probably too colloquial, but I'll be interested to hear this debated.
I'll close this comment for the moment, based on the assumption that you will pick it up in a future version as you explore in your proposed work in defining use cases.

barbaricyawps commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
+* It provides guidance for both template authors as well as document authors.
+* “Template author and template reviewer tips” should be removed from the final about-{doctype}-template.md. Other tips should remain.
+* Tweak other tips to align with the doctype you are writing for.
+
```
I feel like I'm missing the forest for the trees on this document (which is how I felt about the previous version as well). Perhaps what is missing is an overview statement at the beginning that orients me about the overall purpose and use of this document. If I understand this correctly (again, really not sure), the purpose of this template is to guide future template creators on The Good Docs Project. With that in mind, here's a very, very tentative paragraph that you are welcome to change to your heart's content:

```
The base template is for use in The Good Docs Project. Before using this template, see <link to our Contributing Guide, maybe calling out the sections that give and overview define the process for submitting draft templates>. This template outlines the stylistic conventions and overall structure for creating a new Good Docs Project template.
```

Another comment: A good template guides the user in the process of making decisions about how to adapt and use the template for their specific context. I feel like an About template should provide this kind of direction. I'm not sure how to represent that here, but that's what I feel like I'm missing. Sorry that's very vague. I want to think about it some more.

Last comment: The real test will be to see if someone could take this base template and use it effectively to create a new template. Let's consider piloting that kind of test and keeping what Erin calls "a friction log" of where the template helps and hinders.

Loquacity replied:
Couldn't agree more. Because this template is so meta, I think we need to spend the time to explain the point of it. Curse of knowledge means once you understand it, it's hard to remember what was so hard to understand about it, but it took me a good few weeks of planning and research to really work it out, so I think we need to go easy on our readers here.

barbaricyawps reviewed (changes requested):
Please add an overview statement like I've indicated unless you disagree with that suggestion. My other comments are probably out of scope of this PR and could be spun out into a separate issue for later work.

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
+
+* This about-base-template captures the structure and common elements that all downstream about-xxx-templates should follow.
+* It provides guidance for both template authors as well as document authors.
+* “Template author and template reviewer tips” should be removed from the final about-{doctype}-template.md. Other tips should remain.
```
```suggestion
* "Template author and template reviewer tips" should be removed from the final about-{doctype}-template.md. Other tips should remain.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+**Documents:**
+
+* xxx-template.md: Common structure, elements and writing tips for downstream xxx-template.md files.
+* about-xxx-template.md: Common structure, elements and writing tips for downstream about-xxx-template.md files.
```
```suggestion
* xxx-template.md: Common structure, elements, and writing tips for downstream xxx-template.md files.
* about-xxx-template.md: Common structure, elements, and writing tips for downstream about-xxx-template.md files.
```

Loquacity replied:
I wonder if "downstream" is going to be well-understood in this context?

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+### Doctype templates set
+
+Doctype templates provide instructions for document authors to write documentation for a specific doctype.
```
Too many "doc-"s.

```suggestion
Doctype templates provide instructions for writing content for a specific doctype.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+        * How the template might be customized.
+        * When you would or would not use this doctype-template.
+* Example-{doctype}.md:
+    * A fleshed out example of the template, describing the fake chronolog example project. _{TBD we haven’t finished defining this fake project yet.}_
```
```suggestion
    * A more comprehensive example of the template, describing the fake chronologue example project. _{TBD we haven’t finished defining this fake project yet.}_
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+{Template author tip: 
+
+* Discuss the nature of the content for this doctype.
+* Consider how it will impact maintenance.
```
```suggestion
* Consider how the document will be maintained. This can include human work-hours, as well as ongoing infrastructure and hardware requirements.
```

camerons replied:
Thanks @Loquacity , I've merged a variant of this suggestion on top of Felicity's comment.

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+{Template author tip: 
+
+* Each doctype addresses different use cases. It target’s different user personas, with different needs.
+* As such, there are different weightings applied to quality characteristics for each doctype. Reference documentation needs to comprehensively address all the features in the latest version, but can tolerate lower quality writing. Tutorials need only cover a sample of use cases of an older version of the software, but there is a greater need for writing quality.}
```
```suggestion
* Each doctype has its own qualities. Reference documentation needs to comprehensively address all the features in the latest version, but can tolerate lower quality writing. Tutorials only need to cover a sample of use cases of an older version of the software, but there is a greater need for writing quality.}
```

Loquacity replied:
(I'm not sure I agree with this, btw.)

camerons replied:
Thanks @Loquacity I've plucked part of your feedback into the branch.
If you are up for it, I'd be interested to hear you expand on your comment "I'm not sure I agree with this, btw."

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+{Docset owner tip: Select the quality criteria which will be important for your project.}
+
+The following table lists the relative importance of the following quality criteria for {doctype} documents.
```
```suggestion
This table lists the relative importance of the following quality criteria for {doctype} documents.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+|Maintainability: Written to be easily translated.||
+|Maintainability: Uses timeless language with reduced need to update over time.||
+
+Note: The terms “Must”, “Should”, and “May” should be interpreted as per [RFC 2119.](https://www.ietf.org/rfc/rfc2119.txt) 
```
lol, and I thought I was a tech writing theory nerd :wink:

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+* There are often multiple approaches to consider when implementing a doctype.
+* In this section you should discuss different approaches that can be applied to this doctype, including logic and recommendations to help a docset owner select the best strategy for them.
+* Often there would be a light version of this template to use for immature projects, and a more comprehensive version for mature projects. You should discuss how a project might mature their docs from light to comprehensive versions.
```
I would argue that newer projects require more comprehensive docs, because you're starting from zero.

camerons replied:
Valid point. The point I was thinking is more about the capacity of the project to produce documentation, rather than the need for the docs. I've tweaked text to:

* Often there would be a light and more comprehensive version of this template. Immature projects may only have the capacity to maintain the light version. You should discuss how a project might mature their docs from light to comprehensive versions.

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+
+## Maintenance strategy
+
+The long term health of a project’s documentation depends upon the project's ability to establish sustainable maintenance processes for each doctype.
```
```suggestion
The long term health of documentation depends upon the project's ability to establish sustainable maintenance processes for each doctype.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+* Automated reports which highlight:
+    * When documentation is out of date with software.
+    * When documentation hasn’t been reviewed for a while.
+* Integrating documentation checklists into software deployment, build, and test processes.
```
Definitely need to add in here something about deprecation strategies (for pages and books), and also things like how often to perform line-edits vs comprehensive tech/peer reviews vs obsolescence.

camerons replied:
@Loquacity , I think these are strategies which are applied at a "Whole Docset" level. As such they would fit within a "Documentation Management Plan" or maybe within the "Information Architecture Guide".
If you think it fits here, feel free to suggest words to add.
I'll close out this comment for the moment (mainly because I can't think of how to address this in this doc at the moment). We can add into a future iteration.

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+    * Increased developer efficiency.
+    * Faster onboarding for users and/or developers.
+    * Reduced support costs.
+* Consider the case where the docs are only partially maintained. Would it be better to have no documentation that partially maintained documentation?
```
```suggestion
* Consider the case where the docs are only partially maintained. Would it be better to have no documentation than partially maintained documentation?
```


maybe instead of 'partially maintained' it could be worded as 'technically incorrect' or 'out of date'? this makes the tradeoff a little more obvious.

camerons replied:
Good suggestion, added "incorrect".

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
-- An overview of the template you are creating.
-- Prerequisites, the minimum requirements for accessing or  making use of the template.
-- Procedures, step-by-step instructions to create your template.
+* Add a summarised list of reasons for including the doctype, based on the business case above. 
```
```suggestion
* Add a summarized list of reasons for including the doctype, based on the business case above. 
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
-Sandra, the Technical Writer for JKL goes about creating a template for this requirement. She identifies that such a Feature document is a combination of technical reference related to the developed feature, and a how-to guide instructing the user on how to activate the specific feature. The existing templates are not likely to cover this scope fully, and therefore she creates a new How to Add a  ...
+{Template author tip:
 
+* Add a summarised list of reasons for not including the doctype, based on the business case above. 
```
```suggestion
* Add a summarized list of reasons for not including the doctype, based on the business case above. 
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
-- Multiple documents each highlighting a different product, but with a similar level of information on the integrated solution.
-- Different levels of engagement and competencies by the readers. For example, the SOP will be accessed by a developer to integrate a component different to his, or by a systems administrator trying to ensure all servers are running optimally, or by the end user trying to configure the product at his end.
+* The document has been reviewed:
+    * For clear writing and alignment with the writing style guide, ideally by a technical writer.
```
```suggestion
    * For clear writing and alignment with the writing style guide, ideally by a technical writer or editor.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
-- Different levels of engagement and competencies by the readers. For example, the SOP will be accessed by a developer to integrate a component different to his, or by a systems administrator trying to ensure all servers are running optimally, or by the end user trying to configure the product at his end.
+* The document has been reviewed:
+    * For clear writing and alignment with the writing style guide, ideally by a technical writer.
+    * For alignment with the {doctype} template, ideally be a technical writer.
```
```suggestion
    * For alignment with the {doctype} template, ideally by a technical writer, information architect, or content strategist.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+* The document has been reviewed:
+    * For clear writing and alignment with the writing style guide, ideally by a technical writer.
+    * For alignment with the {doctype} template, ideally be a technical writer.
+    * For technical accuracy, ideally by a subject matter expert.
```
```suggestion
    * For technical accuracy, ideally by a technical subject matter expert.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
+{Template author tip:
 
-## Useful References
+* It isn’t sufficient to simply define best practices for a doctype. You need to provide compelling evidence as to why this is the best approach. This section should summarize competing approaches and explain why we recommend the approach taken.
```
```suggestion
* It isn’t sufficient to simply define best practices for a doctype. You need to provide compelling evidence as to why this is the best approach. This section should summarize competing approaches and explain why you took the chosen approach.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -11,84 +24,349 @@ The base template goes one step further, in that it helps you create any templat
 
-## Useful References
+* It isn’t sufficient to simply define best practices for a doctype. You need to provide compelling evidence as to why this is the best approach. This section should summarize competing approaches and explain why we recommend the approach taken.
+* It might suggest situations when a docset owner would select a competing approach.
```
```suggestion
* It might also suggest situations where a docset owner would select a different approach.
```

Loquacity commented on base-template/about-base-template.md:
```diff
@@ -1,8 +1,21 @@
 
+{Template author tip:
+
+* This about-base-template captures the structure and common elements that all downstream about-xxx-templates should follow.
```
There's probably an argument for wrapping the titles in markup of some description to make sentences like this easier to read. If we do, it will need to be repo-wide, though.

camerons replied:
@Loquacity can you please suggest how this should be implemented. I'm unclear on what you are recommending here.
Yes, I'm expecting that after we agree on the base template, we will retrospectively go and update all our templates to align with the base template. (This would be a good first time issue for people wanting to join our project, and wanting to learn our process and how to use git.)

Loquacity replied:
Probaby the best thing to do is to use single backticks. Like this:

```suggestion
* This `about-base-template` captures the structure and common elements that all downstream `about-xxx-templates` should follow.
```

That way, the titles display as monospaced font, which offsets it from the rest of the sentence. Usually, the initcaps would do this work for us, but since we're using all lowercase, I think the markup works to indicate that it's a title of a template.

camerons commented on base-template/base-template.md:
```diff
@@ -1,77 +1,123 @@
 
-1. Technical documentation types - to be able to identify the type of documentation you need to create a template for, or modify the existing one into.</li>
-2. Writing technical documentation - to accurately and clearly express instructions to the reader.</li>
+* All templates must include snippets of [structured data](https://developers.google.com/search/docs/guides/sd-policies). to help search engines and facilitate content reuse. 
```
```suggestion
* All templates should include snippets of [structured data](https://developers.google.com/search/docs/guides/sd-policies). to help search engines and facilitate content reuse. 
```

camerons commented:
Direct message feedback:

From: ...N:  Tue, Jan 12, 5:54 AM 
For the doc base template "about" -- it isn't very approachable. My gut says that you have at least two documents here, maybe three . The intro, prereqs and why/who should be in the about. Then for the rest, I'm not sure -- it seems to be a mix of talk about the base template, boilerplate, and comments to the person using the base template.

How do the comments in the about compare to the actual base-template.md? Maybe you are actually doing a template for a new template's "about" file?

I'd say any template should be just the boilerplate and relevant comments for filling out each section (plus one at the top sending them to another document for the general information.)

Does that help? I may have missed the point but figured that'd be useful info too.

camerons commented:
Feedback sent to me directly. I've [snipped] out information which could be considered personal or project specific.

From: A...
Date: Thu, Jan 14, 2021 at 12:01 PM

Hi Cameron,

Sorry for the delayed response. I wasn't sure how to best send you feedback and don't use GitHub enough to not have to relearn how to do something every time I do it.

These are the three general feelings that I have on reading the template.

*   I'm firmly on board with the idea of giving people, both internally and externally, more guidance about how to write different types of docs.
*   The base template feels very abstract to me and somewhat difficult to understand. I'm concerned that it will feel too complex to your audience, or audiences, and that will deter adoption of or adherence to the templates. I would like for this to all feel simpler.
*   I am opposed to the idea of a fake example project being used to demonstrate how the template is effective.

Some more detailed thoughts:

*   To me, the about-base-template document feels like a guide that explains how to create a template, and I wonder whether it should just be that. If so, my gut feeling is that every template should have a minimal set of common elements:
    *   an optional set of microdata (or whatever you want to call it)
    *   an introduction that identifies the audience and states the purpose of the document
    *   a set of content elements, where the template either dictates or recommends the order in which elements should appear
    *   best practices that a document creator should follow when writing a document
    *   advice for a docset owner trying to figure out whether a document of that type is needed
    *   criteria to use when either evaluating the quality of the finished document or updating/maintaining the document
*   I'm not sure I understand the distinction between the core templates and the doctype templates. Are we anticipating that there will be many implementations of concept templates and reference templates, such that the core (abstract) template contains the common elements of a concept template, and each doctype template is an actual concept template that writers are supposed to use?
*   If you're adopting JSON-LD for structured data, then perhaps consider using the term "linked data" rather than microdata, structured data, etc.
*   In my opinion, there are a variety of criteria that make for a good document and few documents will make use of all of them. Indeed, in some cases, trying to adopt all of the best practices for a doctype might be detrimental if you're forcing yourself to write to the template and not to the audience. As a result, I'm concerned that any documents for a fake project will seem contrived and that their inherent nature will compromise their utility. A couple of years ago, I worked on a project to explain best practices for writing different types of docs that are in the [snip] family of templates. [snip]

The approach I took in writing [snip] wasn't to create a document that followed the template to the letter. Instead, I tried to identify several documents of a particular doctype that followed some or all of the best practices for the associated doctype template. In some cases, the docs didn't even use the template even though they still followed the best practice. I had a couple of goals in taking this approach. First, to put together [snip], I had to look at a lot of different documents [snip], which helped me to understand and recognize common characteristics about good docs. Second, this approach focused on finding good examples of writing a particular type of document and not on finding a good example of a completed template, which I think is different. In so doing, I think we push the notion that the most important thing is following the best practices, not just filling in the template.

This approach might add a long-term maintenance cost in terms of making sure that the good examples you're citing stay good over time. I remember in one case, I had cited the landing page at [snip] as being a really great introduction, but then the [snip] team rewrote that page and added in a bunch of information that shouldn't have been on the landing page, so I had to remove that as an example. However, I think there's a trade-off here. I think a template author should be trying to understand what makes for a good concept/how-to/etc. document and that having that understanding will help them create a better template. So, at the cost of some long-term maintenance, I think this approach will yield a better finished product.

That's my first take here. Happy to discuss more and take a second pass, but I'd like to hear what you think about these ideas first. Feel free to reply over email or [snip].

camerons commented:
* I've applied a major update, addressing lots of review comments (thanks).
* I've reduced the number of target personas, to make the doc easier to read.
* I've split out about-template into base-doctype-guide-template and main-doctype-author-guide, which should make the docs more logical and easier to follow.
* I've added a doctype-example-guide, which is mostly a stub at the moment.
* I've added in a couple more sections which I've been working out as I've been going along.
* I've simplified the description of doc inheritance.
* And more things which I can't remember at the moment.
